umm-spec-lib/src/cmr/umm_spec/migration/characteristics_data_type_migration.clj

```diff
@@ -0,0 +1,73 @@
+(ns cmr.umm-spec.migration.characteristics-data-type-migration
+  "Contains helper functions for migrating between different versions of UMM characteristics' data type."
+  (:require
+   [clojure.set :as set] 
+   [clojure.string :as string]
+   [cmr.common.util :as util :refer [update-in-each]]
+   [cmr.umm-spec.models.umm-common-models :as umm-cmn]
+   [cmr.umm-spec.util :as umm-spec-util]))
+
+(def mapping-up
+  "Defines mappings of Characteristics' data type values from v1.9 to v1.10.
+   Anything not in this map, and not equal to \"NOT APPLICABLE\", map it to \"STRING\"."
+  {"TIME/DIRECTION (ASCENDING)" "STRING"
+   "TIME/DIRECTION (DESCENDING)" "STRING"
+   "VARCHAR" "STRING"
+   "INTEGER" "INT"
+   "RADIOCARBON DATES" "STRING"
+   "STRING" "STRING"
+   "FLOAT" "FLOAT"
+   "INT" "INT"
+   "BOOLEAN" "BOOLEAN"
+   "DATE" "DATE"
+   "TIME" "TIME"
+   "DATETIME" "DATETIME"
+   "DATE_STRING" "DATE_STRING"
+   "TIME_STRING" "TIME_STRING"
+   "DATETIME_STRING" "DATETIME_STRING"})
+
+(defn migrate-data-type
+  "Migrate the Charateristics' data type from string to enum."
+  [characteristics]
+  (when-let [data-type (:DataType characteristics)]
+    (let [upper-case-data-type (string/upper-case data-type)]
+      (when-not (= umm-spec-util/NOT-APPLICABLE upper-case-data-type)    
+       (assoc characteristics :DataType (get mapping-up upper-case-data-type umm-spec-util/STRING))))))
```

Comment:
So we drop any Characteristics that has a DataType of "NOT APPLICABLE"? Just want to make sure.

Reply:
Yes.

Reply:
Wouldn't it make more sense to just use `STRING` rather than dropping the entire Characteristics?

Reply:
No Idea. @eereiter confirmed that we remove the whole Characteristic if the DataType is "NOT APPLICABLE".